### ISSUE TITLE
RMV Provider: Trams and Busses are labelled wrong

### ISSUE THREAD

**Issue by Adwirawien** [public-transport/hafas-client#191]
So as I am experimenting with the RMV provider I noticed two things that looked weird:

Whenever there is a Bus departure, the product is actually "tram" and the mode "train".
And furthermore the trams are labelled with the product "u-bahn".
This is the case for many stops, but the following snippets are for Stop ID 3000010 - Frankfurt (Main) Hauptbahnhof.
Btw. I am using hafas-client 5.7.0 with hafas-rest-api 3.4.0 - the latest versions as of now

Bus departure example:
```json
    {
        "tripId": "1|252418|0|80|2092020",
        "stop": {
            "type": "stop",
            "id": "3000010",
            "name": "Frankfurt (Main) Hauptbahnhof",
            "location": {
                ...
            },
            "products": {
                ...
            }
        },
        "when": "2020-09-02T18:55:00+02:00",
        "delay": 60,
        ...
        "direction": "Frankfurt (Main) Baseler Platz",
        "line": {
            "type": "line",
            "id": "bus-64",
            "fahrtNr": "7861",
            "name": "Bus 64", <---
            "public": true,
            "adminCode": "VGFIC_",
            "mode": "train", <---
            "product": "tram", <---
            "operator": {
                "type": "operator",
                "id": "in-der-city-bus",
                "name": "In der City Bus"
            }
        },
        "remarks": [
            ...
        ]
    },
```

Tram departure example:
```json
   {
        "tripId": "1|21233|0|80|2092020",
        "stop": {
            "type": "stop",
            "id": "3000010",
            "name": "Frankfurt (Main) Hauptbahnhof",
            "location": {
                ...
            },
            "products": {
                ...
            }
        },
        "when": "2020-09-02T18:56:00+02:00",
        "delay": 120,
        ...
        "direction": "Frankfurt (Main) Heilbronner Straße",
        "line": {
            "type": "line",
            "id": "tram-21",
            "fahrtNr": "13137",
            "name": "Tram 21", <---
            "public": true,
            "adminCode": "TRAFBT",
            "mode": "train", <---
            "product": "u-bahn", <---
            "operator": {
                "type": "operator",
                "id": "stadtwerke-verkehrsgesellschaft-frankfurt",
                "name": "Stadtwerke Verkehrsgesellschaft Frankfurt"
            }
        },
        "remarks": [
            ...
        ]
    },
```

**Comment by derhuerst** [comment]
Thanks for reporting!

> Whenever there is a Bus departure, the product is actually "tram" and the mode "train".
> And furthermore the trams are labelled with the product "u-bahn".

This is probably because the [product bitmasks](https://github.com/public-transport/hafas-client/blob/da01544b42aa02bef57906ad5ef4f3e1b456732e/p/rmv/products.js) are wrong. **Would you mind using the ["finding the product bitmasks" guide](https://github.com/public-transport/hafas-client/blob/da01544b42aa02bef57906ad5ef4f3e1b456732e/docs/writing-a-profile.md#finding-the-right-values-for-the-bitmasks-field) and submitting a PR the fixes them?**

---

## side note: debugging incorrectly parsed HAFAS responses

You can debug "raw" HAFAS responses by using the [`debug-cli` command-line helper tool](https://github.com/public-transport/hafas-client/blob/master/tools/debug-cli/cli.js) with the `DEBUG=hafas-client` environment variable. You will see the raw HAFAS request & response (as one line of JSON each) and the equivalent data as parsed by `hafas-client` (pretty-printed across multiple lines).

```shell
env DEBUG=hafas-client ./tools/debug-cli/cli.js rmv departures 3000010 | jq
```

Pick a departure from the parsed data, let's assume the first for this example. You'll find the departure in the raw HAFAS response at `.svcResL[0].res.jnyL[0]`. It references a product from the list at `.svcResL[0].res.common.prodL` via its `prodX` field.

**Comment by Adwirawien** [comment]
Alright, I will try my best to debug this 😁
PR incoming soon I guess

**Comment by Adwirawien** [comment]
Btw.
Another way to get bitmasks easily is to just go on the local transport providers website and inspect the icons from e.g. the departure monitor. The icon link will almost certainly include a number which is the bitmask. However this way may should be used with caution 😄

**Comment by derhuerst** [comment]
Fixed as of 9848dfa76238ff5e625671b077b355bbd596fab2.
